specifyweb/workbench/upload/treerecord.py

```diff
@@ -157,10 +361,25 @@ def _handle_row(self, must_match: bool) -> UploadResult:
             return UploadResult(match_result, {}, {})
 
     def _to_match(self) -> List[TreeDefItemWithParseResults]:
+        def has_non_null_values(parse_results):
+            return any(
+                value is not None
+                for result in parse_results
+                for value in result.filter_on.values()
+            )
+
+        def get_parse_results(tdi):
+            tree_rank_record = TreeRankRecord(tdi.name, tdi.treedef_id)
+            return self.parsedFields.get(tree_rank_record, [])
```

Comment:
Ugh, no. Please, just do a self.parsedFields[tree_rank_record] instead of get. You really gotta stop with all this defensive programming man. Yes, I can see that it'll always be there, but wouldn't you sleep well when you know that the code doesn't have undefined behavior? Isn't the absensce of tree_rank_record in self.parsedFields indicator of a bug somewhere else? Heck this should be an assert -- which you can enforce by using self.parsedFields[tree_rank_record]. 


You think you are creating a "robust" system but you are instead making a blunder where code works when it isn't supposed to (unless, right now, you can detail what happens in all the cases where tree_rank_record isn't present) -- especially to other parts.